java-checks-test-sources/src/main/java/checks/PseudoRandomCheck.java

```diff
@@ -42,12 +46,24 @@ void fun() {
     RandomStringUtils randomStringUtils = new RandomStringUtils();
     String rand8 = randomStringUtils.random(1); // Noncompliant
     String rand9 = RandomStringUtils.random(1); // Noncompliant
+    rand9 = random(1); // Noncompliant
+    rand9 = RandomStringUtils.random(1, 0, 0, false, false, null); // Noncompliant [[sc=31;ec=37]]
+    // Here we should raise an issue only on the `new Random`, not on the call to `random` itself
+    rand9 = RandomStringUtils.random(1, 0, 0, false, false, null, new Random()); // Noncompliant [[sc=71;ec=77]]
+    // Here there should be no issue because `random` will use the supplied secure source of randomness
+    rand9 = RandomStringUtils.random(1, 0, 0, false, false, null, new SecureRandom()); // Compliant
 
     // org.apache.commons.lang3.RandomStringUtils
     org.apache.commons.lang3.RandomStringUtils randomStringUtils2 = new org.apache.commons.lang3.RandomStringUtils();
     String rand10 = randomStringUtils.random(1); // Noncompliant
     String rand11 = org.apache.commons.lang3.RandomStringUtils.random(1); // Noncompliant
+    rand11 = org.apache.commons.lang3.RandomStringUtils.random(1, 0, 0, false, false, null); // Noncompliant [[sc=57;ec=63]]
+    // Here we should raise an issue only on the `new Random`, not on the call to `random` itself
+    rand11 = org.apache.commons.lang3.RandomStringUtils.random(1, 0, 0, false, false, null, new Random()); // Noncompliant [[sc=97;ec=103]]
+    // Here there should be no issue because `random` will use the supplied secure source of randomness
+    rand11 = org.apache.commons.lang3.RandomStringUtils.random(1, 0, 0, false, false, null, new SecureRandom()); // Compliant
 
+    String rand12 = random(42).toLowerCase(Locale.ROOT); // Noncompliant [[sc=21;ec=27]]
```

Comment:
This is a fixed FN - previously we didn't report on statically imported methods because we used the receiver expression to determine the class of the method.